### PR TITLE
Update test-markdown.md

### PR DESCRIPTION
**Pull request by esethna**
1) Remove codeblocking for automated testing
2) Add more markdown tests based on Markdown.md documentation

### CODE REVIEW

File: doc/developer/tests/test-markdown.md [hunk 1]
```diff
@@ -1,25 +1,70 @@
...
+
+### Block Quotes
+
+>This should render in a block quote
+
+### Bulleted Lists
+
+- Item one
+* Item two
+  - Sub-intem one
```
Comment (it33) [inline doc/developer/tests/test-markdown.md]: Typo intem

File: doc/developer/tests/test-markdown.md [hunk 2]
```diff
@@ -1,25 +1,70 @@
...
 ```
 
-Here is how it should render: 
+### In-line Code
+
+The word `monospace` should render as in-line code
+
+### In-line Images
+
+This in-line image should render the mattermost/platform build status:  [![Build Status](https://travis-ci.org/mattermost/platform.svg?branch=master)](https://travis-ci.org/mattermost/platform)
```
Comment (it33) [inline doc/developer/tests/test-markdown.md]: Please try variations of in-line images with image sizes and file sizes more likely to cause issues with rendering so we can better test corner cases?

**Comment by it33** [comment]
Looking really good for a first round of testing, propose adding tests for cases where we've had bugs and regressions previously, e.g. 

1. Not properly rendering carriage return inside of code blocks
2. Rendering text formatting inside of code blocks, 
3. Having carriage returns between bulletted and numbered lists not render as expected
4, Having large in-line images lazy load and cause stutter in the UI

It may be that corner cases for markdown need to get broken out into multiple posts.

**Comment by esethna** [comment]
Closing to submit new PR with many changes
